bin/sync.js

```diff
@@ -202,6 +202,8 @@ function syncRepositoryMilestones(repoOwner, repoName, milestoneDefs, done) {
             console.log('do not create a closed milestone %j', milestone);
             next();
           }
+        } else if(definition === null) {
+          github.issues.deleteMilestone(milestone, cb('delete'));
```

Comment:
This will IMO fail when the milestone does not exist. This happens when the JSON definition contains a `null` milestone, you add a new repository and run the sync. See lines L191-204 for more details.